flink-connectors/flink-connector-kafka-base/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumerBase.java

```diff
@@ -515,15 +510,17 @@ public void open(Configuration configuration) throws Exception {
 			}
 
 			if (filterRestoredPartitionsWithDiscovered) {
-				subscribedPartitionsToStartOffsets.entrySet().removeIf(
-					entry -> {
-						if (!allPartitions.contains(entry.getKey())) {
-							LOG.warn("{} is removed from subscribed Partitions.", entry.getKey());
-							return true;
-						}
-						return false;
+				Iterator<Map.Entry<KafkaTopicPartition, Long>> iter = subscribedPartitionsToStartOffsets.entrySet().iterator();
```

Comment:
Couldn't this be simply:
```
if (filterRestoredPartitionsWithDiscovered) {
    subscribedPartitionsToStartOffsets.entrySet().removeIf(
        entry -> !topicsDescriptor.isMatchingTopic(entry.getKey().getTopic())
    );
}
```
?

Reply:
Done.

Reply:
Just another (unrelevant) question: is there some way to verify the code faster ? I'll have to do `mvn clean verify` locally and it takes a lot of time. 

What is your general workflow of doing this? Thanks :)

Reply:
@fengli If you just want to check the style, you could use `mvn clean validate -Denforcer.skip -T 1C -nsu`

Reply:
@fengli if by verify you also mean running tests, I usually just leave Travis to do this.
Or, if the change is quite a localized change, I sometimes also just run the relevant tests in the IDE.

Reply:
Thanks guys. I installed checkstyle in intellij and use that for style checking. The only drawback is the plugin doesn't have the same version as used by mvn. But I guess that's just a minor issue, most of the time it'll just work.

For the travis test, is it possible to cancel or retrigger a test in case of flink master build failure?